### PR TITLE
[Feat] optimize blob submission data

### PR DESCRIPTION
**Pull request by thedarkjester**
This PR implements issue(s) #

### Checklist

* [x] I wrote new tests for my new core changes.
* [x] I have successfully ran tests, style checker and build against my new changes locally.
* [x] I have informed the team of any breaking changes if there are any.

### CODE REVIEW

File: contracts/contracts/interfaces/l1/ILineaRollup.sol [hunk 1]
```diff
@@ -113,12 +96,12 @@ interface ILineaRollup {
...
    * This value will be used along with the stored last finalized L2 calculated message number in the public input.
    * @dev l2MerkleTreesDepth is the depth of all l2MerkleRoots.
-   * @dev l2MerkleRoots is an array of L2 message Merkle roots of depth l2MerkleTreesDepth between last finalized block and finalSubmissionData.finalBlockInData.
+   * @dev l2MerkleRoots is an array of L2 message Merkle roots of depth l2MerkleTreesDepth between last finalized block and finalSubmissionData.finalBlockNumber.
    * @dev l2MessagingBlocksOffsets indicates by offset from currentL2BlockNumber which L2 blocks contain MessageSent events.
    */
   struct FinalizationDataV3 {
     bytes32 parentStateRootHash;
-    uint256 finalBlockInData;
+    uint256 finalBlockNumber;
```
Comment (jpnovais) [inline contracts/contracts/interfaces/l1/ILineaRollup.sol]:
```suggestion
    uint256 endBlockNumber;
```

`final` leads me to the finality concept, so I would favor interval terminology, start/end

File: contracts/contracts/LineaRollup.sol [hunk 2]
```diff
@@ -69,8 +70,9 @@ contract LineaRollup is
 
   /**
    * @dev NB: THIS IS THE ONLY MAPPING BEING USED FOR DATA SUBMISSION TRACKING.
+   * @dev NB: NB: This was shnarfFinalBlockNumbers and is replaced to indicate only that a shnarf exists with a value of 1.
    */
-  mapping(bytes32 shnarf => uint256 finalBlockNumber) public shnarfFinalBlockNumbers;
+  mapping(bytes32 shnarf => uint256 exists) public shnarfExists;
```
Comment (jpnovais) [inline contracts/contracts/LineaRollup.sol]:
```suggestion
  mapping(bytes32 shnarf => uint256 exists) public blobShnarfExists;
```

File: contracts/contracts/interfaces/l1/ILineaRollup.sol [hunk 3]
```diff
@@ -319,14 +269,14 @@ interface ILineaRollup {
...
    */
-  error SnarkHashIsZeroHash();
+  error FinalShnarfWrong(bytes32 expected, bytes32 value);
 
   /**
-   * @dev Thrown when the computed shnarf does not match what is expected.
+   * @dev Thrown when a parent shnarf does not exist.
    */
-  error FinalShnarfWrong(bytes32 expected, bytes32 value);
+  error ParentShnarfDoesNotExist(bytes32 shnarf);
```
Comment (jpnovais) [inline contracts/contracts/interfaces/l1/ILineaRollup.sol]:
```suggestion
  error ParentBlobNotSubmmitted(bytes32 shnarf);
```

I think this is more indicative of the underlying cause.

Comment (thedarkjester) [reply contracts/contracts/interfaces/l1/ILineaRollup.sol]: addressed in new commit

File: contracts/contracts/LineaRollup.sol (hunk 2 above)
Comment (thedarkjester) [reply contracts/contracts/LineaRollup.sol]: addressed in new commit

File: contracts/contracts/interfaces/l1/ILineaRollup.sol (hunk 1 above)
Comment (thedarkjester) [reply contracts/contracts/interfaces/l1/ILineaRollup.sol]: addressed in new commit

**Review by jpnovais — APPROVED** [review]

**Review by jpnovais — APPROVED** [review]

**Pull request merged**